### ISSUE TITLE
Elaborator scripts appear to not require the extension enabled

### ISSUE THREAD

**Issue by ShinKage** [idris-lang/Idris2#672]
Elaborator scripts are supposed to be run only when the `%language ElabReflection` extension is enabled. Currently, `%runElab` can be called without the the language directive in the module even if it appears that the check is in the source code (See [TTImp.Elab.RunElab:182](https://github.com/idris-lang/Idris2/blob/master/src/TTImp/Elab/RunElab.idr#L182)).

# Steps to Reproduce
Try to typecheck this example:
```idris
import Language.Reflection

testElab : Elab ()
testElab = fail "Should not display this message"

%runElab testElab
```
# Expected Behavior
```
1/1: Building Test (Test.idr)
Error: %language ElabReflection not enabled

Test.idr:6:1--6:18
   |
 6 | %runElab testElab
   | ^^^^^^^^^^^^^^^^^
```

# Observed Behavior
```
1/1: Building Test (Test.idr)
Error: Error during reflection: Should not display this message

Test.idr:6:1--6:18
   |
 6 | %runElab testElab
   | ^^^^^^^^^^^^^^^^^
```

**Comment by MarcelineVQ** [comment]
Seems to be because %runElab at the top level and inside some other definition are somewhat distinct. IRunElabDecl vs IRunElab  
At the top level `processRunElab` is called rather than `checkRunElab` and is missing that language pragma check.

**Comment by ShinKage** [comment]
Thanks for the tip, I've opened a PR with a quick fix.

**Comment by gallais** [comment]
fixed by #674
